File: src/main/scala/pl/multiplex/models/TicketType.scala
package pl.multiplex.models

object TicketType extends Enumeration {

  val ADULT = TicketTypeVal(25, "zł")
  val STUDENT = TicketTypeVal(18, "zł")
  val CHILD = TicketTypeVal(12.5, "zł")

  protected case class TicketTypeVal(price: Double, currency: String) extends super.Val()
}

File: src/main/scala/pl/multiplex/models/TicketType.java
package pl.multiplex.models;

public enum TicketType {

  ADULT(25d, "zł"),
  STUDENT(18d, "zł"),
  CHILD(12.5, "zł");

  private Double price;
  private String currency;

  TicketType(Double price, String currency) {
    this.price = price;
    this.currency = currency;
  }

  public Double getPrice() {
    return price;
  }


}


